test/onnx/onnx_test_common.py

```diff
@@ -436,12 +436,15 @@ def _compare_pytorch_onnx_with_ort(
         ref_input_args = input_args
         ref_input_kwargs = input_kwargs
 
-    # ONNXProgram holds a reference (not copy) to the original ref_model, including its state_dict.
+    # NOTE: ONNXProgram holds a reference (not copy) to the original ref_model, including its state_dict.
     # Thus, ONNXProgram() must run before ref_model() to prevent ref_model.forward() from changing the state_dict.
     # Otherwise, the ref_model can change buffers on state_dict which would be used by ONNXProgram.__call__()
+    # NOTE: `model_with_state_dict=ref_model` is specified to cover runs with FakeTensor support
     ort_outputs = onnx_program(*input_args, model=ref_model, **input_kwargs)
     ref_outputs = ref_model(*ref_input_args, **ref_input_kwargs)
-    ref_outputs = onnx_program.adapt_torch_outputs_to_onnx(ref_model, ref_outputs)
+    ref_outputs = onnx_program.adapt_torch_outputs_to_onnx(
+        ref_outputs, model_with_state_dict=ref_model
```

Comment:
hmm why still need to pass ref_model when it is part of onnx_program init args?

Reply:
LOL From PR description "This is needed because when the model is fakefied, a version of the non-fakefied model is needed so that the Initializers, buffers and constants can be extracted (and used as input to the ONNX model)."

Reply:
still confused .. 2 lines above same `ref_model` is passed as init arg to onnx_program. How is that one fake while the one on L446 non fake

Reply:
wait ok that one is not init, but `__call__` ...